### PR TITLE
[Bug] Address UI Test Failures 

### PR DESCRIPTION
**Pull request by phileo99**
## Purpose

* Revert hardcoded sizes in widgets in SetupControlBarView as flagged by pipeline failures. 
* Fix UI Test to bypass API 31 devices
* make UI Test more flexible and robust
* modify script to auto detect whether physical device is connected or not

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
./gradlew clean cDAT
```

## What to Check
Verify that the following are valid
* all UI tests pass

### CODE REVIEW

**Review by ahammer — APPROVED** [review]

**Comment by ahammer** [comment]
Approved, after merging please cherry-pick to release branch.

**Review by Mrayyan — APPROVED** [review]

**Pull request merged**
